include/net/net_if.h

```diff
@@ -213,106 +345,22 @@ struct net_if {
 	struct net_offload *offload;
 #endif /* CONFIG_NET_OFFLOAD */
 
-#if defined(CONFIG_NET_IPV6)
-#define NET_IF_MAX_IPV6_ADDR CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT
-#define NET_IF_MAX_IPV6_MADDR CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT
-#define NET_IF_MAX_IPV6_PREFIX CONFIG_NET_IF_IPV6_PREFIX_COUNT
-	struct {
-		/** Unicast IP addresses */
-		struct net_if_addr unicast[NET_IF_MAX_IPV6_ADDR];
-
-		/** Multicast IP addresses */
-		struct net_if_mcast_addr mcast[NET_IF_MAX_IPV6_MADDR];
-
-		/** Prefixes */
-		struct net_if_ipv6_prefix prefix[NET_IF_MAX_IPV6_PREFIX];
-
-		/** Router solicitation timer */
-		struct k_delayed_work rs_timer;
-
-		/** Default reachable time (RFC 4861, page 52) */
-		u32_t base_reachable_time;
-
-		/** Reachable time (RFC 4861, page 20) */
-		u32_t reachable_time;
-
-		/** Retransmit timer (RFC 4861, page 52) */
-		u32_t retrans_timer;
-
-		/** IPv6 hop limit */
-		u8_t hop_limit;
-
-#if defined(CONFIG_NET_IPV6_DAD)
-		/** IPv6 current duplicate address detection count */
-		u8_t dad_count;
-#endif /* CONFIG_NET_IPV6_DAD */
-
-		/** RS count */
-		u8_t rs_count;
-	} ipv6;
-#endif /* CONFIG_NET_IPV6 */
-
-#if defined(CONFIG_NET_IPV4)
-#define NET_IF_MAX_IPV4_ADDR CONFIG_NET_IF_UNICAST_IPV4_ADDR_COUNT
-#define NET_IF_MAX_IPV4_MADDR CONFIG_NET_IF_MCAST_IPV4_ADDR_COUNT
-	struct {
-		/** Unicast IP addresses */
-		struct net_if_addr unicast[NET_IF_MAX_IPV4_ADDR];
-
-		/** Multicast IP addresses */
-		struct net_if_mcast_addr mcast[NET_IF_MAX_IPV4_MADDR];
-
-		/** Gateway */
-		struct in_addr gw;
-
-		/** Netmask */
-		struct in_addr netmask;
-
-		/** IPv4 time-to-live */
-		u8_t ttl;
-	} ipv4;
-#endif /* CONFIG_NET_IPV4 */
-
-#if defined(CONFIG_NET_DHCPV4)
-	struct {
-		u32_t xid;
-
-		/** IP address Lease time */
-		u32_t lease_time;
-
-		/** IP address Renewal time */
-		u32_t renewal_time;
-
-		/** IP address Rebinding time */
-		u32_t rebinding_time;
-
-		/** Server ID */
-		struct in_addr server_id;
-
-		/** Requested IP addr */
-		struct in_addr requested_ip;
-
-		/** Timer for DHCPv4 Client requests (DISCOVER,
-		 * REQUEST or RENEWAL)
-		 */
-		struct k_delayed_work timer;
-
-		/** T1 (Renewal) timer */
-		struct k_delayed_work t1_timer;
+} __net_if_align;
```

Comment:
is it still needed though the struct is way much smaller now?

Reply:
It seems we can leave this alignment away, at least the unit test runs fine without the huge alignment.